src/ui/toolbox/menu.js

```diff
@@ -148,16 +172,18 @@ var menuComponentApi = {
             // fromLast (default) navigation: focus on button and then using UP go to last item
             this.hoverIndex = this.menuItems.length - 1; // we start on the button, not at the max array index
             // which would be menuItems.length-1
-            if (this.$menuItems[this.hoverIndex]) {
-                this.$menuItems[this.hoverIndex].focus();
+            const elem = this._findClosestVisibleItem(-1);
+            if (elem) {
+                this.hoverItem(this.menuItems[this.hoverIndex].id);
```

Comment:
This code is not clear since `_findClosestVisibleItem()` is not a pure function. It is not obvious the function also modify the index.

A good implementation would be to have a function returning the index of the first visible elment instead, and then assign the value to the `hoverIndex` property.

Reply:
it just a helper to avoid code duplications. Probably we can return index and move updating hoverIndex to the hoverItem method , but this may affect existing code.

Reply:
why do you want to update the `hoverItem` method? Can't you do something like this?:
```javascript
const index = this.findFirstVisibleItem();
if (index >= 0) {`
    this.hoverIndex = index;
    this.hoverItem(this.menuItems[this.hoverIndex].id);
}
```

Reply:
Ok. if you insist, I'll do it as you suggest. But this code snippet will be duplicated in several places. This is one of the thing I tried to avoid, when moved the reusable code to the method.